robottelo/config/base.py

```diff
@@ -514,6 +514,45 @@ def validate(self):
         return validation_errors
 
 
+class GCESettings(FeatureSettings):
+    """Google Compute Engine settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(GCESettings, self).__init__(*args, **kwargs)
+        self.project_id = None
+        self.client_email = None
+        self.cert_path = None
+        self.zone = None
+        self.cert_url = None
+
+    def read(self, reader):
+        """Read GCE settings."""
+        self.project_id = reader.get('gce', 'project_id')
+        self.client_email = reader.get('gce', 'client_email')
+        self.cert_path = reader.get('gce', 'cert_path')
+        self.zone = reader.get('gce', 'zone')
+        self.cert_url = reader.get('gce', 'cert_url')
+
+    def validate(self):
+        """Validate GCE settings."""
+        valid_cert_path = '/usr/share/foreman/'
+        valid_zones = [
+            'us-east1-b', 'us-east1-c', 'us-east1-d',
+            'us-east4-a', 'us-east4-b', 'us-east4-c']
+        validation_errors = []
+        if not all(self.__dict__.values()):
+            validation_errors.append(
+                'All [gce] {} options must be provided'.format(self.__dict__.keys()))
+        if not str(self.cert_path).startswith(valid_cert_path):
+            validation_errors.append(
+                '[gce] cert_path - cert should be available '
+                'from satellites {}'.format(valid_cert_path))
+        if str(self.zone).startswith('us') and self.zone not in valid_zones:
```

Comment:
So, if I set `zone` to `sfdafasf`, or don't set it at all, I won't get any validation error?

As @vijay8451 , I think we should just check if `self.zone` is in `valid_zones`. `valid_zones` means "zones valid from Robottelo point of view". They can be subset of all zones valid from Google point of view.

Reply:
> So, if I set `zone` to `sfdafasf`, or don't set it at all, I won't get any validation error?

If you dont set the validation above will catch it and ask you to dont keep it empty and the zone has to be provided.

> As @vijay8451 , I think we should just check if `self.zone` is in `valid_zones`. `valid_zones` means "zones valid from Robottelo point of view". They can be subset of all zones valid from Google point of view.

Here, Let's take an example, there is a zone called 'asia-south1-a' actually present in GCE. And I have not listed that one in my valid_zones. So, according to the condition ``` if self.zone in valid_zones``` this becomes an invalid zone, which is not invalid zone at all.

We definitely don't validate zones/cloud properties like ```sfdafasf``` anywhere in our automation unless we have all them known or listed explicitly.

I have three options to choose guys:
1. Keep validation only for 'us' and as it is.
2. List all the valid zones from google.
3. Remove the zones validation at all.

Reply:
> > So, if I set `zone` to `sfdafasf`, or don't set it at all, I won't get any validation error?
> 
> If you dont set the validation above will catch it and ask you to dont keep it empty and the zone has to be provided.

Ah, it's caught by line 543. Alright.

> Here, Let's take an example, there is a zone called 'asia-south1-a' actually present in GCE. And I have not listed that one in my valid_zones. So, according to the condition ` if self.zone in valid_zones` this becomes an invalid zone, which is not invalid zone at all.

On the other hand, with current code zone called 'asia-south10-z' will make it through, while this is something that looks like it could be a real zone, but is not.

For me, letting invalid value through is more serious problem than rejecting value that is technically valid, but never used by the project.

Anyway, looking at https://cloud.google.com/compute/docs/regions-zones/ , I see they have 20 regions, each one with three zones (and one with four zones). Do we know how often regions and zones change? Is this something that we would have to maintain often, or can we just update it every year, or twice a year?

If maintenance burden is low, I would vote for listing all valid zones. `robottelo/constants.py` seems like good place to hold that data.

Personally, I would go as far as generating constant data structure dynamically to ease maintenance and save typing. Something like that:
```python
__zones = {'us-west1': ['a', 'b', 'c'], 'europe-west1': ['b','c','d']}  # and others
VALID_GCE_ZONES = [f'{loc}-{zone}' for loc, zones in zones.items() for zone in __zones]
```